src/integration/java/io/pinecone/helpers/IndexManagerSingleton.java

```diff
@@ -0,0 +1,211 @@
+package io.pinecone.helpers;
+
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.configs.PineconeConfig;
+import io.pinecone.exceptions.PineconeException;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import org.openapitools.client.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.pinecone.helpers.BuildUpsertRequest.buildRequiredUpsertRequestByDimension;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class IndexManagerSingleton {
+    private static final Logger logger = LoggerFactory.getLogger(IndexManager.class);
+    private static IndexManagerSingleton instance;
+    private static final String apiKey = System.getenv("PINECONE_API_KEY");
+    private final int dimension = System.getenv("DIMENSION") == null
+            ? 4
+            : Integer.parseInt(System.getenv("DIMENSION"));
+    private final String environment = System.getenv("PINECONE_ENVIRONMENT") == null
+            ? "us-east4-gcp"
+            : System.getenv("PINECONE_ENVIRONMENT");
+    private static final IndexMetric metric = System.getenv("METRIC") == null
+            ? IndexMetric.COSINE
+            : IndexMetric.valueOf(System.getenv("METRIC"));
+    private static final PineconeConfig pineconeConfig = new PineconeConfig(apiKey);
+    private static Pinecone pineconeClient;
```

Comment:
I don't think `pineconeConfig` needs to be a member var

I don't think `pineconeClient` needs to be static.